### PR TITLE
chore(openapi): update sidebar overview link to be anchor

### PR DESCRIPTION
**Pull request by zchsh**
## 🔗 Relevant links

- [Preview link][preview] 🔎
- [Asana task][task] 🎟️

## 🗒️ What

Adds an `#overview` anchor link to the OpenAPI docs view sidebar, to make it easier for visitors to jump to the top of the page.

## 🧪 Testing

- [ ] Visit [/hcp/api-docs/vault-secrets]
    - The first link in the sidebar should now lead to and `#overview` anchor link
    - The `#overview` anchor link should jump to the top of the page
    - The "active state" on this link should now change when `activeSection` changes, such that only one sidebar link is ever highlighted at one time.

## 💭 Anything else?

Not at the moment!

[task]: https://app.asana.com/0/1204678746647847/1205211659139665/f
[preview]: https://dev-portal-git-zsopenapi-overview-link-hashicorp.vercel.app/
[/hcp/api-docs/vault-secrets]: https://dev-portal-git-zsopenapi-overview-link-hashicorp.vercel.app/hcp/api-docs/vault-secrets

### CODE REVIEW

**Review by braydenlove — APPROVED** [review]
Awesome work dude!

File: src/views/open-api-docs-view/server.ts [hunk 1]
```diff
@@ -66,6 +66,7 @@ export async function getStaticProps({
 	versionData,
 	basePath,
 	statusIndicatorConfig,
+	topOfPageSlug = 'overview',
```
Comment (thiskevinwang) [inline src/views/open-api-docs-view/server.ts:69]: nit: "hash" or "id" might be more appropriate than "slug" here

**Comment by zchsh** [comment]
@thiskevinwang Great question - in this case I think the rationale for the `#overview` ID (good nit btw) relates more closely to the "active state highlighting" in the sidebar. I imagine there's probably some way to refactor `useActiveSection` to support some kind of "top of page" logic without an additional anchor link and injected ID, but it seemed like the simplest way to get the active state highlighted accurately in the sidebar 🤔

**Comment by thiskevinwang** [comment]
@zchsh cool cool, feel free to roll with what you have 👍 

Leveraging [IntersectionObserver](https://developer.mozilla.org/en-US/docs/Web/API/Intersection_Observer_API) in general feels like a less fragile approach.

Refactoring `useActiveSection` _sounds good_, but I'll acknowledge you'd probably have to work with a good mount of existing complexity 😅 .

**Pull request merged**